LibraryManagement.cpp

```diff
@@ -347,161 +337,137 @@ if(bookCounter<10) {
-        	for(int a=i;a<bookCounter;a++) {
-        		bookMark.books[a]=bookMark.books[a+1];
-			}
-			bookMark.books[9].clearBook();
-			bookMark.decrementBook(bookMark.bookCounter);
-			cout<<"The book was successfully deleted.";
-
-			main();
-		} else {
-			main();
-		}
+    for (int i = 0; i < bookCounter; i++) {
+        if (this->books[i].isEqualTo(ISBN)) {
+            cout << "\nBook found\n\n";
+            cout << "Do you want to delete?\n[1]Yes\n[2]No\nEnter choice: ";
+            cin >> choice;
+            if (choice == 1) {
+                this->books[i].clearBook();
+                for(int a = i; a < bookCounter; a++) {
+                    this->books[a] = this->books[a+1];
+                }
+    this->books[19].clearBook();
+    this->decrementBook();
+    cout << "The book was successfully deleted.";
+    menu.displayMainMenu();
+            } else {
+                menu.displayMainMenu();
+              }
+        }
     }
+        cout << "\nBook not found.";
+        menu.displayMainMenu();
 }
-    cout<<"\nBook not found.";
-    main();
-
-
-
-}
-void Bookmark::editBook(int bookCounter) {
-
-    string ISBN, title, author, publisher, categories;
-    string editIsbn,choice;
+void Bookmark::editBook() {
+    string ISBN, title, author, publisher, categories,editIsbn,choice;
     cout << "EDIT BOOK\n\n";
-if(bookCounter==0) {
-	cout<<"There is no book to edit.\n\n";
-	main();
-}
+    if (bookCounter == 0) {
+        cout<<"There is no book to edit.\n\n";
+        menu.displayMainMenu();
+    }
     cout << "Enter ISBN: ";
     cin >> editIsbn;
-
-for (int i = 0; i < bookCounter; i++) {
-    if (bookMark.books[i].getISBN() == editIsbn) {
-    	cout<<"Book found!\n\n";
-    	cout<<"ISBN: "<<bookMark.books[i].getISBN()<<endl;
-    	cout<<"Title: "<<bookMark.books[i].getTitle()<<endl;
-    	cout<<"Author: "<<bookMark.books[i].getAuthor()<<endl;
-    	cout<<"Publisher: "<<bookMark.books[i].getPublisher()<<endl;
-    	cout<<"Categories: "<<bookMark.books[i].getCategories()<<endl;
-        cout<<"\nDo you want to edit?\n[1]Yes\n[2]No\nEnter choice: ";
-        cin>>choice;
-        if(choice=="1") {
-        	while(ISBN.length() != 5) {
-        		cout<<"Enter ISBN: ";
-        		cin>>ISBN;
-
-        		if(ISBN.length() != 5) {
-        			cout<<"ISBN must have 5 digits.\n"<<endl;
-
-				}
-			}
-			cin.ignore();
-			cout<<"-Title: ";
-			getline(cin, title);
-			cout<<"-Author: ";
-			getline(cin, author);
-			cout<<"-Publisher: ";
-			getline(cin, publisher);
-			cout<<"-Categories: ";
-			getline(cin, publisher);
-
-			bookMark.books[i].updateBook(ISBN, title, author, publisher, categories);
-			cout<<"\nThe book has been successfully edited.";
-			main();
-		} else {
-			main();
-		}
+    for (int i = 0; i < bookCounter; i++) {
+        if (this->books[i].getISBN() == editIsbn) {
+            cout << "Book found!\n\n";
+            cout << "ISBN: "<< this->books[i].getISBN() << endl;
```

Comment:
create a method inside book class which will wrote those things and just call that method. For example:

this->books[i].writeDetails()